lang/structs.go

```diff
@@ -638,27 +638,21 @@ func (obj *StmtRes) resource(resName string) (engine.Res, error) {
 			return nil, errwrap.Wrapf(err, "resource field `%s` did not return a type", x.Field)
 		}
 
-		fieldValue, err := x.Value.Value() // Value method on Expr
-		if err != nil {
-			return nil, err
+		name, exists := mapping[x.Field] // lookup recommended field name
+		if !exists {                     // this should be caught during unification. TODO: panic?
```

Comment:
Please remove the `TODO: panic` ... Even if the issue is caused by "programmer error" it's still nicer to normal error and bubble it up with an obscure error message, instead of panicing, and potentially shutting down unsafely. There are a few exceptions though.

Reply:
Sure, removed. I'd hope we can remove most of these checks at some point because a lot of them are redundant